nucypher/policy/conditions/lingo.py

```diff
@@ -291,18 +296,180 @@ def __init__(self, operand: Condition):
 }
 
 
+def _eth_to_wei(a, _):
+    try:
+        return currency.to_wei(a, "ether")
+    except decimal.InvalidOperation as e:
+        raise TypeError(f"Invalid value for ethToWei conversion: {a}") from e
+
+
+def _wei_to_eth(a, _):
+    try:
+        return currency.from_wei(a, "ether")
+    except decimal.InvalidOperation as e:
+        raise TypeError(f"Invalid value for weiToEth conversion: {a}") from e
+
+
+# should raise TypeError for invalid inputs
+_OPERATOR_FUNCTIONS = {
+    # We can add all kinds of operators over time, this is just a base start - given that
+    # we need ethToWei and weiToEth.
+    # Whether this set is too aggressive or not remains to be seen. We can always pare
+    # back on the operators we want to start with.
+    "+=": pyoperator.add,
+    "-=": pyoperator.sub,
+    "*=": pyoperator.mul,
+    "/=": pyoperator.truediv,
+    "%=": pyoperator.mod,
+    "^=": pyoperator.pow,
+    "index": lambda a, b: a[b],
+    "round": lambda a, b: round(a, b),
+    # unary operations i.e. don't require 2nd 'b' value to be passed;
+    # keep extra parameter in lambda function for simplistic execution consistency
+    "abs": lambda a, _: abs(a),
+    "avg": lambda a, _: statistics.mean(a),
+    "ceil": lambda a, _: math.ceil(a),
+    "ethToWei": _eth_to_wei,
+    "floor": lambda a, _: math.floor(a),
+    "len": lambda a, _: len(a),
+    "max": lambda a, _: max(a),
+    "min": lambda a, _: min(a),
+    "sum": lambda a, _: sum(a),
+    "weiToEth": _wei_to_eth,
+    # casting
+    "bool": lambda a, _: bool(a),
+    "float": lambda a, _: float(a),
+    "int": lambda a, _: int(a),
+    "str": lambda a, _: str(a),
+}
+
+_UNARY_OPERATOR_FUNCTIONS = {
+    "abs",
+    "avg",
+    "ceil",
+    "ethToWei",
+    "floor",
+    "len",
+    "max",
+    "min",
+    "sum",
+    "weiToEth",
+    # casting
+    "bool",
+    "float",
+    "int",
+    "str",
+}
+
+MAX_VARIABLE_OPERATIONS = 5
+
+
+class VariableOperation(_Serializable):
+    class Schema(CamelCaseSchema):
+        operation = fields.Str(
+            required=True,
+            validate=OneOf(_OPERATOR_FUNCTIONS, error="Not a permitted operation"),
+        )
+        value = AnyField(required=False, allow_none=True)
+
+        @validates_schema
+        def validate_operation_and_value(self, data, **kwargs):
+            operation = data["operation"]
+            value = data.get("value")
+            if operation in _UNARY_OPERATOR_FUNCTIONS:
+                if value is not None:
+                    raise ValidationError(
+                        field_name="value",
+                        message=f'No value should be provided for operation "{operation}"',
+                    )
```

Comment:
I have a suggestion here. Instead of having a list of operators-without-2nd-value, which sounds to me like code-redundant, we could do something like this. (I have not tested it, though).

```suggestion
from inspect import signature  # at the begginning of the file 

class VariableOperation(_Serializable):
    class Schema(CamelCaseSchema):
        operation = fields.Str(
            required=True,
            validate=OneOf(_OPERATOR_FUNCTIONS, error="Not a permitted operation"),
        )
        value = AnyField(required=False, allow_none=True)

        @validates_schema
        def validate_operation_and_value(self, data, **kwargs):
            operation = data["operation"]
            value = data.get("value")
            if len(signature(operation).parameters) == 1:
                if value is not None:
                    raise ValidationError(
                        field_name="value",
                        message=f'No value should be provided for operation "{operation}"',
                    )
```

Reply:
✔️ - nice! Addressed in https://github.com/nucypher/nucypher/pull/3658/commits/91e373af2ca000748ed766b7bc07a341622753c0